midend/interpreter.cpp

```diff
@@ -716,11 +708,11 @@ void ExpressionEvaluator::postorder(const IR::Operation_Relation* expression) {
         return;
     }
     if (lv->isUnknown()) {
-        set(expression, new SymbolicBool(ScalarValue::ValueState::NotConstant));
+        set(expression, l);
```

Comment:
Why does l have the right type?
If you compare two integers the result is a bool, but l is an integer.
What was wrong with the previous assignment?

Reply:
About this, I can not say anything since it appeared after I merge with the main branch

Reply:
Your merge picked up the wrong changes.
The code you are committing is buggy.
This is a bug I fixed recently. Clearly, expression is a bool, and l may be an int. So this cannot be correct.

Reply:
Why are you resolving this conversation? As I said, this change is wrong.

Reply:
I said 3 times already, this is wrong.